exporter/prometheusexporter/accumulator.go

```diff
@@ -165,8 +165,8 @@ func (a *lastValueAccumulator) accumulateGauge(metric pdata.Metric, il pdata.Ins
 func (a *lastValueAccumulator) accumulateSum(metric pdata.Metric, il pdata.InstrumentationLibrary, now time.Time) (n int) {
 	doubleSum := metric.Sum()
 
-	// Drop metrics with non-cumulative aggregations
-	if doubleSum.AggregationTemporality() != pdata.MetricAggregationTemporalityCumulative {
+	// Drop metrics with non-cumulative aggregations and is not monotonic increasing
+	if doubleSum.AggregationTemporality() != pdata.MetricAggregationTemporalityCumulative && !metric.Sum().IsMonotonic() {
```

Comment:
Isn't it handled by `accumulateGauge`? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e52cfd022fe600d1adf85c4e90f6f0cb4482a988/exporter/prometheusexporter/accumulator.go#L131

Reply:
Since I saw that method and I was looking at the base data model and map:
- [OTLP Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#gauge) -> [Prometheus Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#gauge)
- [OTLP Sums](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#sums) -> [Prometheus Counter](https://prometheus.io/docs/concepts/metric_types/#counter)

So in Prometheus, counters are always monotonic and cumulative. But then I realized it's not a 100% mapped like that, some Sums could be converted back to Prometheus Gauge with convertSum()

```
	case pdata.MetricDataTypeGauge:
		return c.convertGauge(metric)
	case pdata.MetricDataTypeSum:
		return c.convertSum(metric)
```

So maybe we could go with this change below:

```suggestion
	if doubleSum.AggregationTemporality() != pdata.MetricAggregationTemporalityCumulative && doubleSum.AggregationTemporality() != pdata.MetricAggregationTemporalityDelta {
```

With:
```
MetricAggregationTemporalityDelta is a MetricAggregationTemporality for a metric aggregator which reports changes since last report time.
MetricAggregationTemporalityCumulative is a MetricAggregationTemporality for a metric aggregator which reports changes since a fixed start time.
```

Prometheus exporter is at the end of the aggregation (it would set the metric to MetricAggregationTemporalityCumulative in the same function anyway) , I don't think it would care about the time which the metric got reported?

Reply:
More inputs in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5714#issuecomment-1013143984 from @jmacd , I'm reading through his suggestions.

Reply:
Looking through the exporter code here, it seems this code could be modified to accumulate the delta-to-cumulative conversion itself.

Reply:
```suggestion
	if doubleSum.AggregationTemporality() == MetricAggregationTemporalityUnspecified {
```
Better and shorter one.

Reply:
That should never be true.  `MetricAggregationTemporalityUnspecified` is a zero value that should never be set on a real metric and its presence indicates an error.

Reply:
> Looking through the exporter code here, it seems this code could be modified to accumulate the delta-to-cumulative conversion itself.

Can you expand on this?

Reply:
To summarize what I understand from this thread:

When Prometheus receives a Delta temporality counter, there are two cases: (a) monotonic, (b) non-monotonic.

For (a), the treatment can be applied inside Prometheus both easily and correctly. Since this module already keeps state on every metric, performing the delta-to-cumulative is simple. Just keep tallying the sum and it will be cumulative when Prometheus sees it. 

For (b), I will speculate that no one cares and "correct" is difficult to achieve. These are non-monotonic deltas, which statsd already has a _different_ way to treat. If the user sends gauges prefixed by `+` or `-` they have the same semantics as UpDownCounter, are to be interpreted as deltas. So, if the statsd user wanted delta-to-gauge, they already have a way to do that and it's the `statsdreceiver`'s responsibility to compute it (and it does).

Continuing case (b), if the statsd user didn't want delta-to-gauge, then they're presumably sending you positive-and-negative increments. **We have no information about reset time**, so there are two correct ways to interpret this: (a) keep them as deltas, (b) convert them to rate gauges over some time period.  Either way, Prometheus users will still have trouble working with this data, which is why I speculate that no one cares. 

The preceding paragraph lends support to the following claim: Statsdreceiver should by default assume counters are monotonic, because there's a mechanism in statsd for treating non-monotonic counters as gauges already.

Reply:
I think I understand what you're proposing with respect to monotonic, delta temporality counters and agree that we could likely handle the delta-to-cumulative conversion here since this exporter is already maintaining state.  Since https://github.com/open-telemetry/opentelemetry-specification/pull/2266/files#diff-0efae13f08f98e62a81767d5daeff37ebb7ef8c50537c7b9013e72506e9b055aR1085 would specify a different treatment for non-cumulative, monotonic OTLP sum data points being converted to prometheus exposition, do we need to change that or create an exception for the collector, or is there some scoping of that (proposed specification that I'm missing that makes it irrelevant to the collector?

Reply:
Good question. I will comment in https://github.com/open-telemetry/opentelemetry-specification/pull/2266

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/2266#discussion_r787237754

Reply:
Found the document https://opentelemetry.io/docs/reference/specification/metrics/datamodel/#sums-delta-to-cumulative that exactly stated what @jmacd said :D I believe it's 100 the right implementation. Let me see what I could do.

Thanks again for all the great inputs here folks :rock: